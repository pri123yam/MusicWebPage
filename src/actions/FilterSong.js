export const FilterSongs=(filterType,selectedValue,isChecked)=>{
    return{
        type:'FILTER_SONGS',
        payload:{
            filterType:filterType,
            selectedValue:selectedValue,
            isChecked:isChecked
        }
    }
}