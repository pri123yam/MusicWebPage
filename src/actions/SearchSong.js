export const searchSongs = (searchKeyword)=>{
    return{
        type:'SEARCH_SONGS',
        payload:{
            searchKeyword:searchKeyword
        }
    }
}