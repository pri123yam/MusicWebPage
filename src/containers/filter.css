.Filterbuttons{
    background-color: aquamarine;
    float:left;
    width: 10%;
    height:216.5vh;
}