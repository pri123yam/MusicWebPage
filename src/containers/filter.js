import React, { Component } from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { filterSongs } from '../actions/FilterSong';
import GetFilters from '../selectors/GetFilters';
import './filter.css';
export class FilterComponent extends Component {
    handleChange = (event) => {
        let {name:filterType, value:selectedValue, checked:isChecked}=event.target;
        this.props.filterSongs(filterType,selectedValue,isChecked);
    }
    render() {
        return (
            <div className='Filterbuttons'>
                {
                    Object.keys(this.props.filters).map((filteredElement) => {
                        return (
                            <div key={filteredElement} className='container'>
                                <div className="dropdown">
                                    <button className="btn btn-primary dropdown-toggle" type="button" data-toggle="dropdown">{filteredElement}
                                        <span className="caret"></span></button>
                                    <ul className="dropdown-menu" >
                                        {
                                            this.props.filters[filteredElement].map((listItem) => {
                                                return (
                                                    <li key={listItem}><label><input id={listItem} type="checkbox" name={filteredElement} value={listItem} onChange={this.handleChange} /> {listItem} </label> </li>
                                                );
                                            })
                                        }
                                    </ul>
                                </div>
                            </div>
                        );
                    })
                }
            </div>
        );
    }
}
function matchStateToProps(state) {
    return {
        filters: GetFilters(state)
    };
}
function matchDispatchToProps(dispatch) {
    return bindActionCreators({ filterSongs }, dispatch);
}
export default connect(matchStateToProps, matchDispatchToProps)(FilterComponent);