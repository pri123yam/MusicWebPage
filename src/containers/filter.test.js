import React from 'react';
import {FilterComponent} from './filter';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import {filterSongs} from '../actions/FilterSong'
Enzyme.configure({
    adapter: new Adapter()
});

describe('<FilterComponent/>', () => {
            let initProps;
            let wrapper;
            let filterSongsMock;
            beforeEach(() => {
                initProps = {
                    filters: {
                        artist: ['one', 'two', 'three'],
                        genre: ['good', 'bad']
                    }}
                filterSongsMock= jest.fn();
            });

            const event={
                target:{
                    name:'genre',
                    value:'good',
                    isChecked:true
                }
            }

            it('should render <FilterComponent/> component correctly', () => {
                    wrapper = Enzyme.shallow(<FilterComponent 
                                                        filters = {initProps.filters}
                                                        filterSongs={filterSongsMock}
                                                    />);                        
                    
                    expect(wrapper.debug()).toMatchSnapshot();
            });

            it('should trigger the filterSongs() correctly',()=>{
                console.log(wrapper.debug());
                wrapper.debug().find('#good').simulate('change',event);
                expect(filterSongsMock).toHaveBeenCalled();
            })
            afterEach(() => {
                    initProps={};
                    filterSongsMock=undefined;
                });
            });