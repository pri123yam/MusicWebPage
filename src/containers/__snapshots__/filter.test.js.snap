// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilterComponent/> should render <FilterComponent/> component correctly 1`] = `
"<div className=\\"Filterbuttons\\">
  <div className=\\"container\\">
    <div className=\\"dropdown\\">
      <button className=\\"btn btn-primary dropdown-toggle\\" type=\\"button\\" data-toggle=\\"dropdown\\">
        artist
        <span className=\\"caret\\" />
      </button>
      <ul className=\\"dropdown-menu\\">
        <li>
          <label>
            <input id=\\"one\\" type=\\"checkbox\\" name=\\"artist\\" value=\\"one\\" onChange={[Function]} />
             
            one
             
          </label>
           
        </li>
        <li>
          <label>
            <input id=\\"two\\" type=\\"checkbox\\" name=\\"artist\\" value=\\"two\\" onChange={[Function]} />
             
            two
             
          </label>
           
        </li>
        <li>
          <label>
            <input id=\\"three\\" type=\\"checkbox\\" name=\\"artist\\" value=\\"three\\" onChange={[Function]} />
             
            three
             
          </label>
           
        </li>
      </ul>
    </div>
  </div>
  <div className=\\"container\\">
    <div className=\\"dropdown\\">
      <button className=\\"btn btn-primary dropdown-toggle\\" type=\\"button\\" data-toggle=\\"dropdown\\">
        genre
        <span className=\\"caret\\" />
      </button>
      <ul className=\\"dropdown-menu\\">
        <li>
          <label>
            <input id=\\"good\\" type=\\"checkbox\\" name=\\"genre\\" value=\\"good\\" onChange={[Function]} />
             
            good
             
          </label>
           
        </li>
        <li>
          <label>
            <input id=\\"bad\\" type=\\"checkbox\\" name=\\"genre\\" value=\\"bad\\" onChange={[Function]} />
             
            bad
             
          </label>
           
        </li>
      </ul>
    </div>
  </div>
</div>"
`;
